cmd/server.go

```diff
@@ -113,14 +133,40 @@ func NewServerCommand(stopCh <-chan struct{}) *cobra.Command {
 						break
 					}
 				}
-				err = ssr.Run(stopCh)
-				if err != nil {
-					if etcdErr, ok := err.(*errors.EtcdError); ok == true {
-						logger.Errorf("Snapshotter failed with error: %v", etcdErr)
+				err = retry.Do(func() error {
+
+					if err != nil {
+						// It comes here if snapshotter fails to create a snapshot. It
+						// could be because the cloud provider was down. We need to re-initialize
+						// the snapshotter with a new client for the cloud provider.
+						// 5 retries with exponential back-off before harakiri.
+						logger.Infof("Re-initializing Snapshotter in case of error...")
+						ssr, err = reinitializeSnapshotterWithSnapstore(ssr, snapstoreConfig)
+						if err != nil {
+							logger.Errorf("Re-initializing Snapshotter failed: %s", err.Error())
+							handler.Status = http.StatusInternalServerError
+							return err
+						}
+					}
+
+					err = ssr.Run(stopCh)
```

Comment:
Just move the `ssr.Run` at start([line 137](https://github.com/gardener/etcd-backup-restore/pull/25/files#diff-1427b112aa72f64adcf839007e954d56R137)) of function and move reinitialisation logic([line 143](https://github.com/gardener/etcd-backup-restore/pull/25/files#diff-1427b112aa72f64adcf839007e954d56R159) to [line 149](https://github.com/gardener/etcd-backup-restore/pull/25/files#diff-1427b112aa72f64adcf839007e954d56R149)) inside [ `else` case at line 159](https://github.com/gardener/etcd-backup-restore/pull/25/files#diff-1427b112aa72f64adcf839007e954d56R159)